controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -116,10 +119,72 @@ func (r *StorageClassRequestReconciler) Reconcile(ctx context.Context, request r
 
 	result, reconcileError = r.reconcilePhases()
 
-	// Apply status changes to the StorageClassRequest
-	statusError := r.Client.Status().Update(r.ctx, r.StorageClassRequest)
+	statusUpdated := false
+
+	// keep updating the status:
+	// 1. If status update fails due to
+	//   1.1 IsNotFound -> exit as the object must be deleted
+	//   1.2 IsConflict -> the cr got updated, fetch the resource again and retry
+	//   1.3 any other reason, we can retry status update
+	// 2. If status update succeeds -> exit
+	statusError := wait.PollUntilContextTimeout(r.ctx, 100*time.Millisecond, 2*time.Second, true, func(_ context.Context) (done bool, err error) {
+		// Apply status changes to the StorageClassRequest
+		statusError := r.Client.Status().Update(r.ctx, r.StorageClassRequest)
+		if statusError != nil {
+			if errors.IsNotFound(statusError) {
+				return true, statusError
+			}
+			if errors.IsConflict(statusError) {
+				tempStorageClassRequest := &v1alpha1.StorageClassRequest{}
+				tempStorageClassRequest.Name = request.Name
+				tempStorageClassRequest.Namespace = request.Namespace
+
+				if err := r.get(tempStorageClassRequest); err != nil {
+					if errors.IsNotFound(err) {
+						r.log.Error(err, "StorageClassRequest resource not found. Ignoring since object must be deleted.")
+						return true, err
+					}
+					r.log.Error(err, "Failed to get StorageClassRequest.")
+					return false, err
+				}
+				r.StorageClassRequest.Status.DeepCopyInto(&tempStorageClassRequest.Status)
+				tempStorageClassRequest.DeepCopyInto(r.StorageClassRequest)
```

Comment:
the tempStorgeClassRequest has the new object (after the conflict error), and the status is stored in r.StorageClassRequest. We need to update the in-memory object to have the same ObjectMeta (Assuming the spec is not changing, as we already have a plan to make spec immutable).
I think I can modify the above code to something like to make sure that ObjectMeta is matching so that we don't have the failure due to conflict in the next reconcile

> tempStorageClassRequest.ObjectMeta.DeepCopyInto(&r.StorageClassRequest.ObjectMeta)